presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java

```diff
@@ -302,10 +302,8 @@ private void copyFileFromContainer(String filename, Path targetPath)
             return Optional.empty();
         }
 
-        try (DockerClient client = DockerClientFactory.lazyClient()) {
-            InvocationBuilder.AsyncResultCallback<Statistics> callback = new InvocationBuilder.AsyncResultCallback<>();
+        try (DockerClient client = DockerClientFactory.lazyClient(); AsyncResultCallback<Statistics> callback = new AsyncResultCallback<>()) {
```

Comment:
Seems like the library is broken. I would expect the `DockerClient.close` to close the callback passed to `client.statsCmd(getContainerId()).exec(callback);`.
Did you maybe report a bug?

Reply:
No. I didn't. And I wouldn't expect that to be fixed like you've described.